src/main/java/de/imi/mopat/controller/ClinicController.java

```diff
@@ -412,4 +494,84 @@ public String removeClinic(@RequestParam(value = "id", required = true) final Lo
         }
         return showClinics(model);
     }
+
+    private List<ClinicConfigurationMapping> processClinicConfigurationMappingDTOS(Clinic clinic,
```

Comment:
This needs some major refactoring, as it is not really readable in this state. 
Divide into multiple functions, add some comments, try to prevent the constant clinic usage, add some empty lines for better distinction of logic blocks, try to prevent single argument line breaks